### PR TITLE
BUG: Sparse creation with object dtype may raise TypeError

### LINKED ISSUE #11633: Pivot to SparseDataFrame: TypeError: ufunc 'isnan' not supported in sparse matrix conversion
I want to convert a DataFrame to SparseDataFrame before pivoting it (when it gets really sparse, see also [this discussion](http://stackoverflow.com/questions/31661604/efficiently-create-sparse-pivot-tables-in-pandas) ). I have a textual key, which I need to keep ("chr"):

    df = pd.DataFrame( list(zip([3,2,4,1,5,3,2],
                 ["chr1", "chr1", "chr1",  "chr1", "chr2", "chr2", "chr3"], 
                [100,100, 100, 200, 1,3,1],
                [True, True, True, False, True, False, True],
                [-1,0,1,3, 0,2,1])) ,
                columns = ["counts", "chr", "pos", "strand", "distance"])
    
    df.iloc[:,1:].dtypes
    Out[]: 
    chr         object
    pos          int64
    strand        bool
    distance     int64
    dtype: object

For this small table it works well with regular `DataFrame`:

    pd.pivot_table(df, index= [ "chr", "pos"], columns= ["strand","distance"], values= "counts").fillna(0)

         strand   False    True       
    distance     2  3    -1  0  1
    chr  pos                     
    chr1 100     0  0     3  2  4
         200     0  1     0  0  0
    chr2 1       0  0     0  5  0
         3       3  0     0  0  0
    chr3 1       0  0     0  0  2
But I need to do it on much larger matrices. So I tried to do following trick:

    dfpiv = pd.pivot_table(pd.SparseDataFrame(df), index= [ "chr", "pos"], columns= ["strand","distance"], values= "counts")

but I am getting:

    TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

Are there any plans to include a functionality option into `pivot` function for automatic conversion into SparseDataFrame?

### PR DESCRIPTION
**Pull request by sinhrks**
closes #11633 
closes #11856

Added basic tests for `groupby` and `pivot_table`.

### CODE REVIEW

File: pandas/sparse/array.py [hunk 1]
```diff
@@ -152,9 +152,17 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
 
         # Create array, do *not* copy data by default
         if copy:
-            subarr = np.array(values, dtype=dtype, copy=True)
+            try:
+                # ToDo: Can remove this error handling when we actually
+                # support other dtypes
```
Comment (jreback) [inline pandas/sparse/array.py]: hmm, maybe this should be more integrated with ``_sanitize_values`` (or that should be enhanced). not a big deal, but feel free to reorg code to make more general / readable code.

File: pandas/sparse/array.py [hunk 2]
```diff
@@ -437,12 +445,12 @@ def count(self):
 
     @property
     def _null_fill_value(self):
-        return np.isnan(self.fill_value)
+        return com.isnull(self.fill_value)
```
Comment (jreback) [inline pandas/sparse/array.py]: import ``isnull, notnull`` at the top

File: pandas/sparse/array.py (hunk 1 above)
Comment (sinhrks) [inline pandas/sparse/array.py]: Can I leave this ATM, because this can be removed entirely after changing ``dtype`` default to ``None`` (#667).

Comment (jreback) [inline pandas/sparse/array.py]: yes - was just bringing it up mainly for future / FYI

**Comment by jreback** [comment]
thanks @sinhrks very nice as usual

side-note (nothing to do with this PR I think, but should investigate at some point).

```
nosetests pandas/sparse
....
/Users/jreback/pandas/pandas/sparse/array.py:52: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  return _wrap_result(name, op(self.sp_values, other),
```

**Pull request closed without merging**